helm/v1beta1/types_helm_chart_repository.go

```diff
@@ -72,17 +71,45 @@ type ConnectionConfig struct {
 	// It is used as a trust anchor to validate the TLS certificate presented by the remote server.
 	// The key "ca-bundle.crt" is used to locate the data.
 	// If empty, the default system roots are used.
-	// The namespace for this config map is openshift-config.
+	// The namespace for this config map is openshift-config for Cluster scope repositoty if not specified by the user.
+	// For ProjectHelmChartRepository the default will be the namespace where the repository is getting instantiated.
 	// +optional
-	CA configv1.ConfigMapNameReference `json:"ca,omitempty"`
+	CA *ConfigMapNameReference `json:"ca,omitempty"`
```

Comment:
you can't make a non-pointer field into a pointer.

Reply:
We are performing further operations on the connection config and hence wanted to refer it in functions in openshift/console.That's why created a new struct called as ConfigMapNameReference in the same package v1beta1 and propose to use that.

Reply:
using value now instead of pointer.